docs/architecture/commitlog/fileformat.rst

```diff
@@ -0,0 +1,63 @@
+ScyllaDB commitlog segment file format
+======================================
+
+**Note:** Commitlog file formats are subject to change between scylla versions. Users should not make assumptions about nor rely on them.
+Commitlog files should *never* be used across ScyllaDB updates. This information is provided mainly for ScyllaDB contributors.
+
+File descriptor structure
+.........................
+
+ScyllaDB commitlog segment files are named with a versioned, time-indexed scheme, as::
+
+        <Prefix><version>-<id>.log
+
+Where `<Prefix>` is application specific, but typically "Commitlog-", or in case of 
+files being recycled "Recycled-Commitlog-", `<version>` is the file format version,
```

Comment:
What do you mean "application specific", isn't the only application we're considering Scylla? Or maybe you are hinting that there are other "applications" like hints? Can you list all these "applications"?
And as a rather stupid reader, I didn't understand the mention of "files being recycled" - what does this mean and why do these exist? Again, this can be part of a more general overview, before the actual format, that I think is missing from this document.

Reply:
Application specific -> application of commitlog functionality inside scylla. I.e. "normal" commitlog, schema commitlog and hints. 
The concept of recycled segments is an aspect of the data life cycle inside the CL logic. Agree it might not be obvious to any reader, but should be explained elsewhere. Also fairly common knowledge amongst those working with/against CL.